### PR TITLE
Don't use `@_semantics`, use our own attribute instead.

### PR DESCRIPTION
**Pull request by grynspan**
This PR replaces our use of `@_semantics` with a custom attribute macro. `@_semantics` is reserved for use by the standard library and runtime and, while useful, can be emulated on our side of the module barrier without much difficulty.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci test

**Comment by grynspan** [comment]
@swift-ci test

**Comment by grynspan** [comment]
@swift-ci test

File: Sources/Testing/Test+Macro.swift [hunk 1]
```diff
@@ -484,6 +484,27 @@ extension Test {
...
+///
+/// The use cases for this macro are subject to change over time as the needs of
+/// the testing library change. The implementation of this macro in the
+/// TestingMacros target determines how different arguments are handled.
+///
+/// - Note: This macro has compile-time effects _only_ and should not affect a
+///   compiled test target.
+///
+/// - Warning: This macro is used to implement the `@Test` macro. Do not use it
+///   directly.
```
Comment (stmontgomery) [inline Sources/Testing/Test+Macro.swift:503]: Strictly speaking, isn't it used to implement by `#expect` macros, not `@Test`? Maybe this should be more vague

File: Sources/Testing/Test+Macro.swift [hunk 2]
```diff
@@ -484,6 +484,27 @@ extension Test {
...
+@attached(peer) public macro __testing(
```
Comment (stmontgomery) [inline Sources/Testing/Test+Macro.swift:504]: Does this _need_ to be `public` initially? I suspect that's the direction this is headed regardless, but just wondering if it's only needed by our own test code right now

File: Sources/TestingMacros/PragmaMacro.swift [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+/// Attributes conditionally applied with `#if` are ignored.
+func pragmas(on node: some WithAttributesSyntax) -> [AttributeSyntax] {
+  node.attributes
+    .compactMap { attribute in
+      if case let .attribute(attribute) = attribute {
+        return attribute
+      }
+      return nil
+    }.filter { attribute in
+      attribute.attributeNameText == "__testing"
```
Comment (stmontgomery) [inline Sources/TestingMacros/PragmaMacro.swift:54]: Is this robust enough to accept a fully-qualified name with module prefix like `@Testing.__testing` ?

Comment (stmontgomery) [reply Sources/TestingMacros/PragmaMacro.swift:54]: (And conversely, to reject attributes with the same name which could hypothetically exist in other modules?)

File: Sources/Testing/Test+Macro.swift (hunk 1 above)
Comment (grynspan) [reply Sources/Testing/Test+Macro.swift:503]: Yes, no, maybe, both :)

Right now it's actually used to implement one of our tests only.

File: Sources/TestingMacros/PragmaMacro.swift (hunk 3 above)
Comment (grynspan) [reply Sources/TestingMacros/PragmaMacro.swift:54]: FBOFW that isn't actually supported by the language, so… no. But I can code it defensively if you're concerned.

Comment (grynspan) [reply Sources/TestingMacros/PragmaMacro.swift:54]: Actually, no, I don't think we care: realistically nobody else is going to write `@__testing(semantics: "...")`. If somebody really _does_ run into trouble here, we can address it in the future.

File: Sources/Testing/Test+Macro.swift (hunk 2 above)
Comment (grynspan) [reply Sources/Testing/Test+Macro.swift:504]: We can make it internal for now—there's a public version coming almost immediately after this PR though.

Comment (grynspan) [reply Sources/Testing/Test+Macro.swift:504]: See https://github.com/swiftlang/swift-testing/pull/745/commits/ddfdbc9a751f167dd4c6c99cb4c949739f813182

File: Sources/TestingMacros/PragmaMacro.swift (hunk 3 above)
Comment (grynspan) [reply Sources/TestingMacros/PragmaMacro.swift]: Added fully-qualified name check. If somebody else makes a macro named `@__testing(semantics:)`, it'll trip as ambiguous before getting to our code, so I don't care too much. :)

**Comment by grynspan** [comment]
@swift-ci test

File: Sources/Testing/Test+Macro.swift (hunk 1 above)
Comment (stmontgomery) [reply Sources/Testing/Test+Macro.swift:503]:
```suggestion
/// - Warning: This macro is used to implement other macros declared by the testing
///   library. Do not use it directly.
```

**Review by stmontgomery — APPROVED** [review]

**Comment by grynspan** [comment]
@swift-ci test

**Pull request merged**